controllers/tempo/tempostack_controller.go

```diff
@@ -82,6 +82,8 @@ func (r *TempoStackReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			log.Error(err, "unable to fetch TempoStack")
 			return ctrl.Result{}, fmt.Errorf("could not fetch tempo: %w", err)
 		}
+		// instance is not found, metrics can be cleared
+		status.ClearTempoStackMetrics(req.Namespace, req.Name)
```

Comment:
do we always get a reconcile event on deletion? could we remove the `ClearTempoStackMetrics()` from the deletion webhook then?